utils/cosmos/iscn/sign.ts

```diff
@@ -89,26 +101,33 @@ function formatISCNPayload(payload: ISCNSignPayload, version = 1) {
     authorNames,
     authorUrls,
     authorWallets,
-  } = payload;
+  } = payload
 
-  const contentFingerprints = [];
-  if (fileSHA256) contentFingerprints.push(`hash://sha256/${fileSHA256}`);
-  if (ipfsHash) contentFingerprints.push(`ipfs://${ipfsHash}`);
-  const stakeholders = [];
+  const contentFingerprints = []
+  if (fileSHA256) contentFingerprints.push(`hash://sha256/${fileSHA256}`)
+  if (ipfsHash) contentFingerprints.push(`ipfs://${ipfsHash}`)
+  const stakeholders = []
   if (authorNames.length) {
     for (let i = 0; i < authorNames.length; i += 1) {
-      const authorName = authorNames[i];
-      const authorUrl = authorUrls[i];
-      const isNonEmpty = authorUrl || authorName;
+      const authorName = authorNames[i]
+      const authorUrl = authorUrls[i]
+      const authorId = authorWallets[i]
+      const isNonEmpty = authorUrl || authorName
       if (isNonEmpty) {
-        stakeholders.push(Buffer.from(JSON.stringify({
-          entity: {
-            id: authorUrl || undefined,
-            name: authorName,
-          },
-          rewardProportion: 1,
-          contributionType: 'http://schema.org/author',
-        }), 'utf8'))
+        stakeholders.push(
+          Buffer.from(
+            JSON.stringify({
+              entity: {
+                '@id': `did:cosmos:${authorId.slice(6)}` || undefined,
```

Comment:
id should not be left as empty
if wallet not exists, use url
if wallet nor url exists, use name as id